### ISSUE TITLE
Parsing error when there are multiple spaces in front of a comment on a block-string

### ISSUE THREAD

**Issue by jandubois** [goccy/go-yaml#550]
I find it a bit hard to describe, but the example below will hopefully clarify it. You can parse a line like `- field: | # comment` (or any other block style indicator, like `|-` etc), but if there is more than one space between the block style indicator and the `#` sign, then there is a parsing error.

Here is a `test.go` program to demonstrate:

```go
package main

import (
	"fmt"
	"github.com/goccy/go-yaml"
)

func main() {
	document1 := `
foo: | # comment
  bar
`
	document2 := `
foo: |  # comment
  bar
`
	var s map[string]string
	if err := yaml.Unmarshal([]byte(document1), &s); err != nil {
		fmt.Print(err)
	} else {
		fmt.Print(s["foo"])
	}
	if err := yaml.Unmarshal([]byte(document2), &s); err != nil {
		fmt.Print(err)
	} else {
		fmt.Print(s["foo"])
	}
}
```

Using Go 1.23.2.

With go-yaml v1.13.6 I get:

```console
$ go run test.go
bar
[2:6] found invalid token
>  2 | foo: |  # comment
            ^
   3 |   bar
```

And with v1.15.2 I get

```console
$ go run test.go
bar
[2:6] invalid header option: " "
>  2 | foo: |  # comment
            ^
   3 |   bar
```

**Comment by goccy** [comment]
Thank you for your reporting ! It is likely a parsing issue with the literal header option.
I'll fix this soon 👍

**Comment by goccy** [comment]
@jandubois  I've fixed this problem with v1.15.3 ! Please confirm it when you have time 🙏

**Comment by jandubois** [comment]
> I've fixed this problem with v1.15.3

Thank you very much! 🙏 

I've confirmed that the problem is gone with the latest release.
